### PR TITLE
year select with range

### PR DESCRIPTION
**Pull request by kafelix496**
#11425 when I select a year with a range, only the start year is highlighted.

### CODE REVIEW

**Comment by kafelix496** [comment]
@cetincakiroglu Hey, I also added a feature to block selection of year if the year is less/greater than min/max year. Please take a look.

File: src/app/components/calendar/calendar.ts [hunk 1]
```diff
@@ -1268,6 +1277,40 @@ export class Calendar implements OnInit,OnDestroy,ControlValueAccessor {
...
+        let between: boolean = false;
+        if (start && start instanceof Date && end && end instanceof Date) {
+            return start.getFullYear() <= year && end.getFullYear() >= year;
+        }
+
+        return between;
+    }
+
+    isLessThanMinYear(year) {
+        if (this.minDate) {
```
Comment (volvachev) [inline src/app/components/calendar/calendar.ts:1295]: I think it's better.
```ts
return this.minDate && year < this.minDate.getFullYear();
```

File: src/app/components/calendar/calendar.ts [hunk 2]
```diff
@@ -1268,6 +1277,40 @@ export class Calendar implements OnInit,OnDestroy,ControlValueAccessor {
...
+          if (year < this.minDate.getFullYear()) {
+            return true;
+          }
+        }
+
+        return false;
+    }
+
+    isGreaterThanMaxYear(year) {
+        if (this.maxDate) {
```
Comment (volvachev) [inline src/app/components/calendar/calendar.ts:1305]: I think it's better.
```ts
return this.maxDate && this.maxDate.getFullYear() < year;
```

File: src/app/components/calendar/calendar.ts (hunk 1 above)
Comment (kafelix496) [reply src/app/components/calendar/calendar.ts:1295]: Oh, yes. You are right. I'll update it. Thanks!

File: src/app/components/calendar/calendar.ts (hunk 2 above)
Comment (kafelix496) [reply src/app/components/calendar/calendar.ts:1305]: Oh, yes. You are right. I'll update it. Thanks!

File: src/app/components/calendar/calendar.ts [hunk 3]
```diff
@@ -1268,6 +1277,40 @@ export class Calendar implements OnInit,OnDestroy,ControlValueAccessor {
         return between;
     }
 
+    isYearEquals(value, year) {
+        if (value && value instanceof Date) return value.getFullYear() === year;
+        else return false;
+    }
+
+    isYearBetween(start, end, year) {
+        let between: boolean = false;
```
Comment (volvachev) [inline src/app/components/calendar/calendar.ts:1286]: redundant variable. `between`
```ts
if () {
return ....
}

return false;
```

File: src/app/components/calendar/calendar.ts [hunk 4]
```diff
@@ -1268,6 +1277,40 @@ export class Calendar implements OnInit,OnDestroy,ControlValueAccessor {
         return between;
     }
 
+    isYearEquals(value, year) {
+        if (value && value instanceof Date) return value.getFullYear() === year;
```
Comment (volvachev) [inline src/app/components/calendar/calendar.ts:1281]: How about this variant code?
```ts
return value && value instanceof Date && value.getFullYear() === year;
```

**Comment by sujeetjaiswara** [comment]
When is this important merge coming?

**Pull request closed without merging**